### PR TITLE
created a AboutUs screen-fixed issue#27

### PR DESCRIPTION
**Pull request by Pranav108**
This Pull Request is about issue #27 ,here I created a AboutUs screen which shows us a brief info about this app and it's contributors along with the working link to their respective profile. I have to use http package in this issue which caused a lot version confliction with firebase packages.
So finally I also updated each dependency to their latest version keeping version confliction, null safety and other upcoming problems in mind. And during solving relative errors I also updated minSdkVersion and targetSdkVersion.
And I haven't attached this screens to any option yet because even after several attempt I didn't find anything on home screen to attach with.

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@Pranav108, was there a template provided when you opened the PR, or was there none? (#44 should have had added a template but I'm not sure if it is working)

File: android/app/build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ if (localPropertiesFile.exists()) {
 
 def flutterRoot = localProperties.getProperty('flutter.sdk')
 if (flutterRoot == null) {
-    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
+    throw new FileNotFoundException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
```
Comment (Saransh-cpp) [inline android/app/build.gradle:11]: Reason for this?

File: lib/Screens/AboutUs.dart [hunk 2]
```diff
@@ -1,10 +1,362 @@
...
+class AboutUs extends StatefulWidget {
+  @override
+  _AboutUsState createState() => _AboutUsState();
+}
+
+class _AboutUsState extends State<AboutUs> {
+  final double spacing = 40;
+  List jsonContributors = [];
+  String _about_this_app =
+      "BookRentApp application is specifically made for renting books.It displays collections of top selling books as well as the option to have your taste curated and matched with a book.Here customers can filter and select books based on genre, after which placing immediate or deferred orders with the option to cancel.";
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:19]:
```suggestion
      "Chapter3 is an app specifically made for renting books. It displays collections of top-selling books as well as the option to have your taste curated and matched with a book. Here customers can filter and select books based on genre, after which placing immediate or deferred orders with the option to cancel.";
```

File: lib/Screens/AboutUs.dart [hunk 3]
```diff
@@ -1,10 +1,362 @@
...
+
+  Future<void> _fetchContributors() async {
+    const contributorsAPIUrl =
+        'https://api.github.com/repos/Saransh-cpp/BookRentApp-Chapter3/contributors';
+
+    //final response = await http.get(Uri(host: contributorsAPIUrl));
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:25]:
```suggestion
```

File: lib/Screens/AboutUs.dart [hunk 4]
```diff
@@ -1,10 +1,362 @@
...
+    final response = await http.get(Uri.parse(contributorsAPIUrl));
+
+    if (response.statusCode == 200) {
+      setState(() {
+        //jsonContributors = json.decode(response.body) as List; //original
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:30]:
```suggestion
```

File: lib/Screens/AboutUs.dart [hunk 5]
```diff
@@ -1,10 +1,362 @@
...
+        jsonContributors = jsonDecode(response.body) as List;
+      });
+    } else {
+      Fluttertoast.showToast(
+        msg: "Failed to load Contributors from API",
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:35]:
```suggestion
        msg: "Failed to load Contributors.",
```

File: lib/Screens/AboutUs.dart [hunk 6]
```diff
@@ -1,10 +1,362 @@
...
+                        children: <Widget>[
+                          const Text(
+                            "ABOUT",
+                            style: TextStyle(
+                                fontSize: 25, fontWeight: FontWeight.w700),
+                          ),
+                          Container(
+                            margin: const EdgeInsets.only(left: 8),
+                            child: const Text(
+                              "BOOK-RENT-APP",
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:84]:
```suggestion
                              "Chapter 3",
```

File: lib/Screens/AboutUs.dart [hunk 7]
```diff
@@ -1,10 +1,362 @@
...
+                          padding: EdgeInsets.all(20),
+                          child: Text(_about_this_app),
+                        ),
+                      ),
+                    ),
+                    Container(
+                      margin: const EdgeInsets.only(top: 15),
+                      alignment: Alignment.topCenter,
+                      child: const Text(
+                        "Presenting The Team",
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:111]:
```suggestion
                        "Presenting the Developers",
```

File: lib/Screens/AboutUs.dart [hunk 8]
```diff
@@ -1,10 +1,362 @@
...
+                    if (jsonContributors.length > 0)
+                      Container(
+                        height: 170,
+                        margin: const EdgeInsets.only(top: 20),
+                        child: ListView.builder(
+                          shrinkWrap: true,
+                          scrollDirection: Axis.horizontal,
+                          itemCount: jsonContributors.length,
+                          itemBuilder: (BuildContext context, int index) =>
+                              (jsonContributors[index]["login"].toString() ==
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:170]: I think you should also skip the `allcontributors` bot here, with my account.

File: lib/Screens/AboutUs.dart [hunk 9]
```diff
@@ -1,10 +1,362 @@
...
+      child: SizedBox(
+        width: 100,
+        height: 100,
+        child: DecoratedBox(
+          decoration: BoxDecoration(
+            boxShadow: _buildBoxShadow,
+            color: Colors.black,
+            shape: BoxShape.circle,
+            image: DecorationImage(
+              //image: AssetImage('images/pranav.jpg'),
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:254]:
```suggestion
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @Pranav108! Looks very good overall. Here is a higher-level review. I'll look into it more soon!

File: android/app/build.gradle (hunk 1 above)
Comment (Pranav108) [reply android/app/build.gradle:11]: GradleExecption was showing in red(as deprecated I guess) , so I searched for its alternative and found FileNotFoundException as its latest alternative.

Comment (Saransh-cpp) [reply android/app/build.gradle:11]: Oh oh, nice work!

**Comment by Pranav108** [comment]
> @Pranav108, was there a template provided when you opened the PR, or was there none? (#44 should have had added a template but I'm not sure if it is working)

I don't think so(or maybe few minutes before) because I was going through this repo whole day and must had noticed this kind of change.
And in hurry I forgot to give a last touch to the code. Should have removed those extra comments, I will take care of this things from next time.

**Comment by Pranav108** [comment]
I'm getting an option of 'resolve conversion' here, so I have to change it or you are improving it?
This is first time I'm going though this things, so don't know how to proceed.

**Comment by Saransh-cpp** [comment]
> I'm getting an option of 'resolve conversion' here, so I have to change it or you are improving it? This is first time I'm going though this things, so don't know how to proceed.

These are some suggestions that should be implemented in this PR. You can do so directly from GitHub using the `Commit suggestion` button.

**Comment by Saransh-cpp** [comment]
@Pranav108, can you merge the master branch of the main repository in this branch? Your branch is 20 commits behind, hence the home screen error (which has been fixed in the master branch).
image

Let me know if you need any help:) !

**Comment by Pranav108** [comment]
> @Pranav108, can you merge the master branch of the main repository in this branch? Your branch is 20 commits behind, hence the home screen error (which has been fixed in the master branch). image
> 
> Let me know if you need any help:) !

I just did it, was that correct?

**Review by Pranav108 — COMMENTED** [review]
I think this should be fine now.

File: lib/Screens/AboutUs.dart [hunk 10]
```diff
@@ -1,10 +1,355 @@
...
+
+class AboutUs extends StatefulWidget {
+  @override
+  _AboutUsState createState() => _AboutUsState();
+}
+
+class _AboutUsState extends State<AboutUs> {
+  final double spacing = 40;
+  List jsonContributors = [];
+  String _about_this_app =
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:18]:
```suggestion
  String _aboutThisApp=
```

File: lib/Screens/AboutUs.dart [hunk 11]
```diff
@@ -1,10 +1,355 @@
...
+
+    final response = await http.get(Uri.parse(contributorsAPIUrl));
+
+    if (response.statusCode == 200) {
+      setState(() {
+        jsonContributors = jsonDecode(response.body) as List;
+      });
+    } else {
+      Fluttertoast.showToast(
+        msg: "Failed to load Contributors.",
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:33]:
```suggestion
        msg: "Failed to load contributors.",
```

File: lib/Screens/AboutUs.dart [hunk 12]
```diff
@@ -1,10 +1,355 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchContributors();
+    Future.delayed(const Duration(seconds: 2), () {
+      setState(() {
+        _isLoading = false;
+      });
+    });
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:50]:
```suggestion
```

I don't think we need this. A constant timer isn't very useful while switching between screens.

File: lib/Screens/AboutUs.dart [hunk 13]
```diff
@@ -1,10 +1,355 @@
...
+    } else {
+      Fluttertoast.showToast(
+        msg: "Failed to load Contributors.",
+        toastLength: Toast.LENGTH_SHORT,
+        gravity: ToastGravity.BOTTOM,
+      );
+    }
+  }
+
+  bool _isLoading = true;
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:40]:
```suggestion
```

File: lib/Screens/AboutUs.dart [hunk 14]
```diff
@@ -1,10 +1,355 @@
...
 
-class AboutUs extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
+      body: _isLoading
+          ? const SpinKitRotatingCircle(
+              color: Colors.blue,
+            )
+          : SafeArea(
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:60]:
```suggestion
      body: SafeArea(
```

As we don't want the timer now.

File: lib/Screens/AboutUs.dart [hunk 15]
```diff
@@ -1,10 +1,355 @@
...
+                                fontSize: 25, fontWeight: FontWeight.w700),
+                          ),
+                          Container(
+                            margin: const EdgeInsets.only(left: 8),
+                            child: const Text(
+                              "Chapter 3",
+                              style: TextStyle(
+                                fontSize: 25,
+                                fontWeight: FontWeight.w700,
+                                // color: Colors.orange[900],
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:86]:
```suggestion
```

File: lib/Screens/AboutUs.dart [hunk 16]
```diff
@@ -1,10 +1,355 @@
...
+                    Container(
+                      alignment: Alignment.topCenter,
+                      margin: EdgeInsets.only(
+                        top: 10,
+                      ),
+                      child: Row(
+                        mainAxisSize: MainAxisSize.min,
+                        children: <Widget>[
+                          const Text(
+                            "ABOUT",
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:75]: Could you add an `AppBar` saying "About" (with a default back arrow)? Then this about can be removed from here.

File: lib/Screens/AboutUs.dart [hunk 17]
```diff
@@ -1,10 +1,355 @@
...
+                            style: TextStyle(
+                                fontSize: 25, fontWeight: FontWeight.w700),
+                          ),
+                          Container(
+                            margin: const EdgeInsets.only(left: 8),
+                            child: const Text(
+                              "Chapter 3",
+                              style: TextStyle(
+                                fontSize: 25,
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:84]:
```suggestion
                                fontSize: 30,
```

File: lib/Screens/AboutUs.dart [hunk 18]
```diff
@@ -1,10 +1,355 @@
...
+                    ),
+                    Container(
+                      padding:
+                          EdgeInsets.symmetric(vertical: 10, horizontal: 15),
+                      child: Card(
+                        elevation: 1,
+                        shadowColor: Colors.grey,
+                        child: Padding(
+                          padding: EdgeInsets.all(20),
+                          child: Text(_about_this_app),
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:101]:
```suggestion
                          child: Text(_aboutThisApp),
```

Can you also add center align here?

File: lib/Screens/AboutUs.dart [hunk 19]
```diff
@@ -1,10 +1,355 @@
...
+          blurRadius: 6,
+        ),
+      ];
+
+  Future<void> _launchURL(String url) async {
+    if (await canLaunch(url)) {
+      await launch(url);
+    } else {
+      print('Could not launch $url');
+      throw 'Could not launch $url';
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:344]: We could add a snack bar here

File: lib/Screens/AboutUs.dart [hunk 20]
```diff
@@ -1,10 +1,355 @@
...
+                    ),
+                    Container(
+                      margin: const EdgeInsets.only(top: 15),
+                      alignment: Alignment.topCenter,
+                      child: const Text(
+                        "Presenting the Developers",
+                        style: TextStyle(
+                            fontSize: 18, fontWeight: FontWeight.w500),
+                      ),
+                    ),
```
Comment (Saransh-cpp) [inline lib/Screens/AboutUs.dart:113]:
```suggestion
```

On second thought, I think presenting the developers looks too weird😅

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @Pranav108! Awesome work! This is the final review -

File: lib/Screens/AboutUs.dart (hunk 19 above)
Comment (Pranav108) [reply lib/Screens/AboutUs.dart:317]: I have added a SnackBar but on multiple calling they are stored in a queue which cause each of them to appear one after another which is annoying ,  tried to fix it by using a flag variable but was unable to fix this. 
Will be if we tap on icon several times but the snackBar appear only once ,I searched about it but didn't find something relevant ,Can you please suggest me how to improve it?

**Comment by Pranav108** [comment]
Thanks for your suggestion, they really improved the view and code for screen a lot.
I also removed some redundant code as far I found them. Let me know about further improvement that need to be added.
This is the layout till now.
AboutUsImprovementSection

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @Pranav108! Looks great now! Merging!

File: lib/Screens/AboutUs.dart (hunk 19 above)
Comment (Saransh-cpp) [reply lib/Screens/AboutUs.dart:317]: I think we can leave it as it is.

**Comment by Saransh-cpp** [comment]
@all-contributors please add @Pranav108 for code, design

**Pull request merged**